src/Common/GWPAsan.h

```diff
@@ -19,12 +19,23 @@ bool isGWPAsanError(uintptr_t fault_address);
 
 void printReport(uintptr_t fault_address);
 
+extern std::atomic<bool> init_finished;
+
+void initFinished();
+
 extern std::atomic<double> force_sample_probability;
 
 void setForceSampleProbability(double value);
 
+inline bool shouldSample()
+{
+    return init_finished.load(std::memory_order_relaxed) && GuardedAlloc.shouldSample();
+}
```

Comment:
Please add a comment explaining why we don't want to sample allocations during startup:
 - to avoid complex issues with initialization order
 - don't waste MaxSimultaneousAllocations on global objects